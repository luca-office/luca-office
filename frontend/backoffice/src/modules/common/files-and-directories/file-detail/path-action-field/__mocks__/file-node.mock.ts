import {FileNode} from "shared/components"
import {BinaryType, NodeType} from "shared/enums"

export const fileNodeMock: FileNode = {
  binaryFileId: "spfosdkfsd",
  binaryFileUrl: "url@url.de",
  binaryType: BinaryType.Image,
  id: "sfpoksdf-fposjfd",
  name: "test-file",
  parentId: "sfopksdf-fsdpfosdk",
  tags: [],
  type: NodeType.File
}
