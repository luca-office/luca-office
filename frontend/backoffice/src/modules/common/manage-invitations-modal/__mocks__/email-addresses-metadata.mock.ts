import {EmailAddressesMetadata} from "../models/email-addresses-metadata"

export const emailAddressesMetadataMock: EmailAddressesMetadata = {
  validAddresses: [
    "testmail@test.de",
    "cap@cap3.de",
    "testmail@test.de",
    "testmai2l@test.de",
    "cap3@cap3.de",
    "cap@cap3.de"
  ],
  uniqueAddresses: ["testmail@test.de", "cap@cap3.de", "testmai2l@test.de", "cap3@cap3.de"],
  hasInvalidAddress: false,
  duplicatedAddresses: ["testmail@test.de"],
  alreadyInvitedAddresses: ["cap@cap3.de"]
}
