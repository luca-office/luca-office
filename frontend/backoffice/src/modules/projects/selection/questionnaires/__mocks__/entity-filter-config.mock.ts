import {EntityFilterOptionState, EntityFilterOptionType, EntitySortOption} from "../../../../../enums"
import {EntityFilterConfig} from "../../../../../models"

export const entityFilterConfigMock: EntityFilterConfig = {
  sortBy: EntitySortOption.BY_NAME,
  state: EntityFilterOptionState.SHOW_ALL,
  type: EntityFilterOptionType.ALL_ENTITIES,
  search: ""
}
