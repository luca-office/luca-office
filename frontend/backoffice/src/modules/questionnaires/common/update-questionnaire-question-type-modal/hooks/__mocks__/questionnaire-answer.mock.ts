import {QuestionnaireAnswer} from "shared/models"

export const questionnaireAnswerMock = (
  questionId: UUID = "18316dac-2355-4e45-a7f3-f339461a488f"
): QuestionnaireAnswer => ({
  __typename: "QuestionnaireAnswer",
  createdAt: new Date(2020, 10, 5).toISOString(),
  modifiedAt: new Date(2020, 10, 15).toISOString(),
  id: "1fce8289-71fa-4568-bccc-329dc4cc67eb",
  text: "answer",
  questionId,
  isCorrect: false,
  position: 1
})
