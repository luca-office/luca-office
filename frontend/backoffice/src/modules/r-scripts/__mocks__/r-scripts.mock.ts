import {RScript} from "shared/models"

export const rScriptMock: RScript = {
  __typename: "RScript",
  id: "18de306c-a8f8-4311-9a2f-ca5418a245bd",
  createdAt: "2020-08-03 15:16:28.501000 +00:00",
  title: "title",
  description: "description",
  modifiedAt: "2020-08-03 15:16:28.501000 +00:00",
  archivedAt: null,
  gitCommitHash: "sd",
  version: "1.2",
  isEditable: true
}

export const rScriptsMock: RScript[] = [
  rScriptMock,
  {...rScriptMock, id: "18de306c-a8f8-4311-9a2f-ca5418a2452d"},
  {...rScriptMock, id: "18de306c-a8f8-4311-9a2f-ca5418a2453d"},
  {...rScriptMock, id: "18de306c-a8f8-4311-9a2f-ca5418a2454d"},
  {...rScriptMock, id: "18de306c-a8f8-4311-9a2f-ca5418a2455d"}
]
