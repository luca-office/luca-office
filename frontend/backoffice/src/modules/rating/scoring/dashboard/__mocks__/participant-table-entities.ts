import {ParticipantTableEntity} from "../scoring-dashboard-table/scoring-dashboard-table"

export const mockParticipantTableEntities: ParticipantTableEntity[] = [
  {
    id: "1",
    isFinalScore: false,
    isRatingCompleted: false,
    isRatingOfMainRater: false,
    name: "name 1",
    index: 1,
    ratingCounts: {ratableProjectModulesCount: 12, finalRatedProjectModulesCount: 6},
    isRatingOfAllModulesPossible: true,
    isRatingOfSomeModulePossible: true
  },
  {
    id: "2",
    isFinalScore: false,
    isRatingCompleted: false,
    isRatingOfMainRater: false,
    name: "name 3",
    index: 2,
    ratingCounts: {ratableProjectModulesCount: 12, finalRatedProjectModulesCount: 6},
    isRatingOfAllModulesPossible: true,
    isRatingOfSomeModulePossible: true
  },
  {
    id: "3",
    isFinalScore: false,
    isRatingCompleted: false,
    isRatingOfMainRater: false,
    name: "name 3",
    index: 3,
    ratingCounts: {ratableProjectModulesCount: 12, finalRatedProjectModulesCount: 6},
    isRatingOfAllModulesPossible: true,
    isRatingOfSomeModulePossible: true
  }
]
