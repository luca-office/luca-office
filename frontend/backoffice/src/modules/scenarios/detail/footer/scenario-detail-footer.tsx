import {css} from "@emotion/react"
import * as React from "react"
import {Button, CardFooterItem} from "shared/components"
import {IconName} from "shared/enums"
import {
  backgroundColorBright,
  CustomStyle,
  Flex,
  spacing,
  spacingHuge,
  spacingLarge,
  spacingMedium
} from "shared/styles"
import {useLucaTranslation} from "shared/translations"
import {formatDateFromString} from "shared/utils"
import {UserAccount} from "../../../../models"

export interface ScenarioDetailFooterProps extends CustomStyle {
  readonly author: UserAccount
  readonly createdAt: string
  readonly openPreview: () => void
}

export const ScenarioDetailFooter: React.FC<ScenarioDetailFooterProps> = ({author, createdAt, openPreview}) => {
  const {t} = useLucaTranslation()

  return (
    <div css={styles.cardFooter}>
      <div css={Flex.row}>
        <CardFooterItem
          title={t("detail_card__title_createdAt")}
          icon={IconName.Calendar}
          text={formatDateFromString(createdAt)}
        />
        <CardFooterItem
          customStyles={styles.author}
          title={t("detail_card__title_author")}
          icon={IconName.Profile}
          text={`${author.firstName} ${author.lastName}`}
        />
      </div>
      <Button onClick={openPreview}>{t("preview")}</Button>
    </div>
  )
}

const styles = {
  cardFooter: css(Flex.row, {
    padding: spacing(spacingMedium, spacingHuge),
    justifyContent: "space-between",
    backgroundColor: backgroundColorBright
  }),
  author: css({
    marginLeft: spacingLarge
  })
}
