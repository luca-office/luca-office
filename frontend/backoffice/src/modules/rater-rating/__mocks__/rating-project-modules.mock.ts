import {projectModulesMockWithQuestionnaire, surveyInvitationsMock} from "shared/graphql/__mocks__"
import {RatingProjectModule} from "../models"

export const ratingProjectModulesMock: RatingProjectModule[] = projectModulesMockWithQuestionnaire.map(
  (projectModule, index) => ({
    ...projectModule,
    ratingsByParticipants: surveyInvitationsMock.map(({id}, index) => ({
      participantId: id,
      isFullyRated: index % 2 === 0,
      isNotRatable: false
    })),
    participantCount: index + 1,
    ratedParticipantCount: index
  })
)
