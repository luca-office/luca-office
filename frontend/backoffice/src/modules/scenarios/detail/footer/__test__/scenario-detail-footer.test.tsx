import {shallow} from "enzyme"
import * as React from "react"
import {create} from "react-test-renderer"
import {Button, CardFooterItem} from "shared/components"
import {userAccountMock} from "shared/graphql/__mocks__"
import {ScenarioDetailFooter} from "../scenario-detail-footer"

describe("scenario-detail-footer", () => {
  const createdAt = new Date(1, 1, 2020).toISOString()
  it("renders correctly", () => {
    const component = create(
      <ScenarioDetailFooter author={userAccountMock} createdAt={createdAt} openPreview={jest.fn()} />
    )

    const tree = component.toJSON()
    expect(tree).toMatchSnapshot()
  })
  it("has correct structure", () => {
    const component = shallow(
      <ScenarioDetailFooter author={userAccountMock} createdAt={createdAt} openPreview={jest.fn()} />
    )
    expect(component.find(CardFooterItem)).toHaveLength(2)
    expect(component.find(Button)).toHaveLength(1)
  })
})
