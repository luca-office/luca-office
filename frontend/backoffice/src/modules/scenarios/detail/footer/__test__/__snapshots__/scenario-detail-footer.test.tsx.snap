// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`scenario-detail-footer renders correctly 1`] = `
<div
  className="css-hwuw3h-ScenarioDetailFooter"
>
  <div
    className="css-14oslo1-ScenarioDetailFooter"
  >
    <div
      className="css-1839z0z-CardFooterItem"
      title="detail_card__title_createdAt"
    >
      <div
        className="css-1lrzij5-Icon"
        dangerouslySetInnerHTML={
          {
            "__html": {},
          }
        }
      />
      <div
        className="css-1de4iva-Text"
      >
        13.08.1906
      </div>
    </div>
    <div
      className="css-nsmsx2-CardFooterItem"
      title="detail_card__title_author"
    >
      <div
        className="css-1lrzij5-Icon"
        dangerouslySetInnerHTML={
          {
            "__html": {},
          }
        }
      />
      <div
        className="css-1de4iva-Text"
      >
        John Doe
      </div>
    </div>
  </div>
  <button
    className="css-injsqh-Button"
    disabled={false}
    onClick={[Function]}
  >
    <div
      className="css-95j2hm-Button"
    >
      preview
    </div>
  </button>
</div>
`;
